src/js/server/apply/logic.js

```diff
@@ -45,3 +45,60 @@ exports.createApplicationFromForm = function (formData, user) {
     
   });
 };
+
+exports.createTeamFromForm = function (formData, user) {
+  const members = new Set();
+  const hackerIds = [];
+  // Ensure application slugs are unique and not the applicant's own
+  return user.getHackerApplication().then(application => {
+    // Start off with the current hacker's application slug — we already know they're in the team
+    members.add(application.applicationSlug);
+  }).then(new Promise((resolve, reject) => {
+    for (const applicationSlug of [formData.memberB, formData.memberC, formData.memberD].map(s => s.trim()).filter(s => s !== '')) {
+      if (!members.has(applicationSlug)) {
+        members.add(applicationSlug);
+      } else {
+        throw new Error('Application slugs must be distinct.');
+      }
+    }
+    resolve();
+  })).then(() => {
+    const applicationSlugs = Array.from(members);
+    if (applicationSlugs.length > 1) {
+      return Promise.all(applicationSlugs.map(applicationSlug => {
+        return HackerApplication.findOne({
+          where: { applicationSlug }
+        }).then(application => {
+          if (application === null) {
+            // The application slug was not valid
+            throw new Error('The application slug matched no hacker.');
+          }
+          return application;
+        }).then(application => {
+          hackerIds.push(application.hackerId);
+          return TeamMember.findOne({
+            where: { hackerId: application.hackerId }
+          }).then(application => {
+            if (application !== null) {
+              // The hacker is already part of another team
+              throw new Error('A team member can\'t belong to more than one team.');
+            }
+          });
+        });
+      }));
+    } else {
+      throw new Error('You need at least two team members to form a team.');
+    }
+  }).then(() => {
+    // Create a new team
+    Team.create({ }).then(team => {
+      // Add the team members to the team
+      for (const hackerId of hackerIds) {
```

Comment:
It's a bit sparse on how to set foreign keys here. I imagined it would set them automatically, as long as they'd been set up, but when I tried it, it didn't seem to be the case (or maybe I just got something a little wrong). I can get it to return a promise anyway, but you're right in that it could potentially leave teams without hackers if something goes wrong. I'll work on the more vital things first, and come back to this if there's time.